core/src/test/scala/org/apache/spark/serializer/JavaSerializerSuite.scala

```diff
@@ -25,4 +25,18 @@ class JavaSerializerSuite extends SparkFunSuite {
     val instance = serializer.newInstance()
     instance.deserialize[JavaSerializer](instance.serialize(serializer))
   }
+
+  test("Deserialize object containing a primitive Class as attribute") {
+    val serializer = new JavaSerializer(new SparkConf())
+    val instance = serializer.newInstance()
+    instance.deserialize[JavaSerializer](instance.serialize(new ContainsPrimitiveClass()))
+  }
```

Comment:
yeah, you're right, I don't see a good way around this.  I'm OK with this in that case, but lets leave this open for a bit to see if anybody else has any ideas